state/indexer/block/kv/kv.go

```diff
@@ -293,10 +294,17 @@ LOOP:
 			continue
 		}
 
-		if _, ok := qr.AnyBound().(int64); ok {
-			v, err := strconv.ParseInt(eventValue, 10, 64)
-			if err != nil {
-				continue LOOP
+		if _, ok := qr.AnyBound().(*big.Int); ok { // This means that in a range we can only have integers, floats will be ignored
+			v := new(big.Int)
+			v, ok := v.SetString(eventValue, 10)
+			if !ok {
+
+				v_f, err := strconv.ParseFloat(eventValue, 64)
+				if err != nil {
+					continue LOOP
+				}
+				v = new(big.Int)
+				v = v.SetInt64(int64(v_f))
```

Comment:
Two questions:
1) This part was not present in the original. Do we need it?
2) Should `ParseFloat` be replaced by a bog float?

Reply:
Hey, no it was not, this case basically covers the functionality we discussed today that we decided to ignore. This is when an event is stored in the DB as a float and it will not be parsed as BigInt (the `ok` in line 298 will be false). To mimic the original behaviour in 0.34 (before introducing BigInts), we need to still somehow parse the float into at least the int64. This same reasoning applies to the change in [this part ](https://github.com/cometbft/cometbft/blob/1dd27a7cb11df703065a29b173ed53b2e3cd374e/libs/pubsub/query/query.go#L476)of the code . Note that there I do use a big float. And I am actually considering now what the best approach is. Because using the BigFloat used in `query.go` rounds the float (which is not done with a simple cast of float64 to int64), but using a simple cast will not parse BigFloats and ints properly. Trying to find a solution and will mark the PR as draft and re-open it to avoid confusion.

Reply:
I think what you suggested in slack would be the simplest (to be consistent with the current behavior in `v0.34.x`/`v0.37.x`): if there is a dot, just split the string and use the whole number part